packages/nextjs/app/debug/_components/contract/utilsContract.tsx

```diff
@@ -49,4 +71,79 @@ const getInitialFormState = (abiFunction: AbiFunction) => {
   return initialForm;
 };
 
-export { getFunctionInputKey, getInitialFormState, getParsedContractFunctionArgs };
+const getInitalTupleFormState = (abiTupleParameter: AbiParameterTuple) => {
+  const initialForm: Record<string, any> = {};
+  if (abiTupleParameter.components.length === 0) return initialForm;
+
+  abiTupleParameter.components.forEach((component, componentIndex) => {
+    const key = getFunctionInputKey(abiTupleParameter.name || "tuple", component, componentIndex);
+    initialForm[key] = "";
+  });
+  return initialForm;
+};
+
+const getInitalTupleArrayFormState = (abiTupleParameter: AbiParameterTuple) => {
+  const initialForm: Record<string, any> = {};
+  if (abiTupleParameter.components.length === 0) return initialForm;
+  abiTupleParameter.components.forEach((component, componentIndex) => {
+    const key = getFunctionInputKey("0_" + abiTupleParameter.name || "tuple", component, componentIndex);
```

Comment:
why "0_"?

Reply:
Actually, this function constructs the initial state for `TupleArray` and since initially while rendering `TupleArray`we are displaying 0th element hence a state for it. 

<details>
<summary>
More detailed explanation 
</summary>

Lets suppose : 

```solidity
struct SimpleStruct {
  uint x;
  uint y;
}

// ---

struct NestedStruct {
  uint a;
  SimpleStruct[] b;
}

```

Then the inital form state  for `b`  `TupleArray` will look like : 

```js
{
    "0_b_x_uint256_uint256": "",
    "0_b_y_uint256_uint256": ""
}
```
since default are showing 0th element

</details>